### ISSUE TITLE
~emo triggers overtake topics

### ISSUE THREAD

**Issue by Lewwwk** [silentrob/superscript#221]
I currently have some `~emohello *(2)` and `~emothanks` triggers in `random. If however, I'm in a topic, those triggers are overriding matches within the topic. 

e.g. 

```
+ test
- {topic=test}

> topic:system:keep test
+ *(1-5)
- responding in test topic
< topic

+ ~emohello
- {keep} responding in random hello

+ ~emothanks [*]
- {keep} responding in random thanks

+ something else
- {keep} replying with something else
```

if I enter `test` i go in the test topic as expected. If I say `something else` i'll get `responding in test topic` as expected.  If I then say `hello` or `thanks`, i'll get `responding in random hello` or `responding in random thanks` which i am not expecting.

**Comment by Lewwwk** [comment]
Did some more testing and what I'm seeing is that it does try to match in my topic first, but the ~emo'd messages don't match.  It then keeps looking, goes in to pre, and then into random where it finds the ~emo response.

```
  Common Try to match (clean)'~emohello' against __welcome__ (__welcome__) +5ms
  Common Try to match (lemma)'~emohello' against __welcome__ (__welcome__) +0ms
  Common Try to match (clean)'~emohello' against (\s?(?:[\w-]*\.?\,?\s?){1,6}) ((\s?(?:[\w-]*\.?\,?\s?){1,6})) +0ms
  Common Try to match (lemma)'~emohello' against (\s?(?:[\w-]*\.?\,?\s?){1,6}) ((\s?(?:[\w-]*\.?\,?\s?){1,6})) +0ms
  Common Try to match (clean)'~emohello' against (\s?(?:[\w-]*\.?\,?\s?){7,99}) ((\s?(?:[\w-]*\.?\,?\s?){7,99})) +0ms
  Common Try to match (lemma)'~emohello' against (\s?(?:[\w-]*\.?\,?\s?){7,99}) ((\s?(?:[\w-]*\.?\,?\s?){7,99})) +0ms
```
In this case the three triggers in my topic are:
```
+ __welcome__
+ *(1-6)
+ *(1-7)
```

A couple of other notes. 

The user was in the `__onboarding__` topic
```
User getTopic +2.5m __onboarding__
```

The topic search order does seem appropriate: 
```
  GetReply Topics to check +2.5m [ { name: '__pre__', type: 'TOPIC', id: 56910273d3f74ee6387681e7 },
  { id: 56910271d3f74ee6387681cc, type: 'REPLY' },
  { name: '__onboarding__',
    type: 'TOPIC',
    id: 56910271d3f74ee6387681ca },
  { id: 56910264d3f74ee638768174,
    name: 'random',
    system: false,
    type: 'TOPIC' },
  { name: '__post__', type: 'TOPIC', id: 56910273d3f74ee6387681e8 } ]
```

And in my case with the `__onboarding__` topic, I have in its own topic file named `_onboarding.ss` to help ensure it gets priority. (realize that shouldn't matter because it is a system topic, and what's happening is that the ~emohello version isn't matching any trigger in the topic).

**Comment by silentrob** [comment]
The problem was with leading emo words or more specifically the leading tilde `~` this was fixed in ss-parser 40c4190168c0b9949310b36906a6aab65ab80c80 and test added here.
